server/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -174,19 +173,26 @@ public long readBlobPreferredLength() {
         return Long.MAX_VALUE;
     }
 
+    private static final StandardOpenOption[] OVERWRITE =
+            new StandardOpenOption[]{StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.CREATE};
+
+    private static final StandardOpenOption[] CREATE_NEW = new StandardOpenOption[]{StandardOpenOption.CREATE_NEW};
+
     @Override
     public void writeBlob(String blobName, InputStream inputStream, long blobSize, boolean failIfAlreadyExists) throws IOException {
-        if (failIfAlreadyExists == false) {
-            deleteBlobsIgnoringIfNotExists(Collections.singletonList(blobName));
-        }
         final Path file = path.resolve(blobName);
-        try (OutputStream outputStream = Files.newOutputStream(file, StandardOpenOption.CREATE_NEW)) {
```

Comment:
This was kind of racy to begin with. It is conceivable that this could fail in a master-failover simulation (and may have been the cause of some past test failures on Windows) if two masters run this concurrently (during finalization) then one of them will fail the write even with `failIfAlreadyExists` set to `false`.